server/session.go

```diff
@@ -103,6 +111,11 @@ type Session struct {
 	// subs concurrently.
 	subsLock sync.RWMutex
 
+	// Map of remote topic subscriptions, indexed by topic name.
+	remoteSubs map[string]*RemoteSubscription
+	// Synchronizes access to remoteSubs.
+	remoteSubsLock sync.RWMutex
+
 	// Cluster nodes to inform when the session is disconnected
 	nodes map[string]bool
```

Comment:
As I said in the previous review, this will likely contain stale entries. It's not horrible, just inefficient.

Reply:
Maybe get rid of it in favor of the `remoteSubs`

Reply:
Just uploaded another commit. Please, take a look.

Reply:
LGTM